### DISCUSSION TITLE
Query Job using BigQueryInsertJobOperator does extra work: After inserting data to a BigQuery partitioned table (has policy tag), it also queries the destination table and lead to errors

### DISCUSSION THREAD

**Discussion by stevenzhang-support** [apache/airflow#22771] [General]
**Issue Summary**:
The main issue is the expected behavior of using or not using **Fine grained reader permission** [1] in the job is different from the actual behavior. From the documentation [2], INSERT statements should not require any **Fine grained reader permission** but controversially the customer's job does require this [3] unexpectedly.

The customer job is not strictly an **INSERT** statement but using _BigQueryInsertJobOperator_ in an Airflow job. This raised a question whether or not _BigQueryInsertJobOperator_ does something more than just INSERT statements as per documentation [2].

The error received [3] only appeared when _BigQueryInsertJobOperator_ is used without **Fine grained reader permission**, but no matter whether the Fine grained reader permission is used or not, the data can be written without any issue. When **Fine grained reader permission** was added in the execution roleset, the error disappeared. 

Question: _BigQueryInsertJobOperator_ does not INSERT data into the table but actually invoked a query on the destination table that is why **Fine grained reader permission** was checked in this operator and caused the error per [3].

[1] https://cloud.google.com/bigquery/docs/column-level-security-intro#roles
[2] https://cloud.google.com/bigquery/docs/column-level-security-writes#inserting_data
[3]: google.api_core.exceptions.Forbidden: 403 GET https://bigquery.googleapis.com/bigquery/v2/projects/rea-gcp-ds-analytics/queries/<airflow-job-ID>?maxResults=0&location=australia-southeast1&prettyPrint=false: User does not have permission to access data protected by policy tag "personal-information-spike : email-address" on column <project-ID>:<dataset-ID>.<table-ID>$20220401.email.

**Comment by stevenzhang-support** [comment]
Please kindly provide some insights

**Comment by potiuk** [comment]
I think the best way to deal with that is to submit PR fixing it if you see the problem.

**Reply by stevenzhang-support** [reply]
Thanks and I am not sure whether or not I should call it as an issue to be fixed. Maybe this is worked as intended but not ideal behavior?

**Reply by potiuk** [reply]
No idea.
